.circleci/config.yml

```diff
@@ -91,85 +90,230 @@ oracle-db-container: &oracle-db-container
-      sudo unzip vendor/oracle/instantclient-odbc-linux.x64-12.2.0.1.0-2.zip -d /opt/oracle
-      sudo ln -s /opt/oracle/instantclient_12_2/libclntsh.so.12.1 /opt/oracle/instantclient_12_2/libclntsh.so
-      sudo rm -rf /opt/system/vendor/oracle
-      rm -rf vendor/oracle/*.zip
-      sudo cp config/oracle/*.ini /etc/
+memcached-container: &memcached-container
+  image: memcached:1.5-alpine
 
-jobs:
-  dependencies_bundler:
-    docker:
-      - *system-builder
+redis-container: &redis-container
+  image: redis:4.0-alpine
+
+dnsmasq-container: &dnsmasq-container
+  image: quay.io/mikz/dnsmasq
+  command:
+  - --no-poll
+  - --address=/#/127.0.0.1
+
+##################################### CIRCLECI COMMANDS ############################################
+
+commands: # reusable commands with parameters
+
+  checkout-with-submodules:
     steps:
       - checkout
-      - *git-submodules
-      - *restore-bundler-cache
+      - run:
+          name: "Pull Submodules"
+          command: |
+            git submodule init
+            git submodule update
+
+  install-gem-dependencies:
+    steps:
       - run:
           name: Install gems with bundler
           command: |
+            echo "Running gems install for database: $DB"
             bundle check || bundle install --deployment
             bundle clean
             # remove capybara-webkit source, save more than 400 MB
             rm -rf "$BUNDLE_PATH/$(ruby -e 'puts Gem.ruby_engine')/$(ruby -e 'puts Gem.ruby_api_version')"/gems/capybara-webkit-*/src
           environment:
             BUNDLE_RETRY: 3
             BUNDLE_JOBS: 3
-      - save_cache:
-          key: v1-bundler-gems-{{ arch }}-{{ .Branch }}-{{ checksum "Gemfile.lock" }}-{{ checksum "gemfiles/prod/Gemfile.lock" }}
-          paths:
-            - ./vendor/bundle
-            - ./.bundle/
-      - persist_to_workspace:
-          root: .
-          paths:
-          - ./vendor/bundle
-          - ./.bundle/
-  deps_bundler_oracle:
-    docker:
-      - *system-builder
+
+  clone-oracle-libraries:
     steps:
-      - checkout
-      - *git-submodules
-      - *restore-bundler-cache
       - add_ssh_keys:
           fingerprints:
           - "36:6a:7a:93:88:52:12:dd:4c:84:3a:42:bf:8c:c3:58"
-      - *clone-oracle-libs
       - run:
-          name: Install gems with bundler
+          name: "Fetch oracle libraries"
           command: |
-            DB=oracle bundle check || DB=oracle bundle install --deployment
-            DB=oracle bundle clean
-            # remove capybara-webkit source, save more than 400 MB
-            rm -rf "$BUNDLE_PATH/$(ruby -e 'puts Gem.ruby_engine')/$(ruby -e 'puts Gem.ruby_api_version')"/gems/capybara-webkit-*/src
+            git clone git@github.com:3scale/system-libraries.git
+            cp -R system-libraries/vendor/oracle/*.zip vendor/oracle
+            sudo unzip vendor/oracle/instantclient-basiclite-linux.x64-12.2.0.1.0.zip -d /opt/oracle
```

Comment:
Good enough.
This is because of https://github.com/3scale/system-builder/blob/1bc3cec26bff04e0603e1a4908594b70a114dfe8/Dockerfile#L16-L17